File: data/frama_c_problems/absolute_value.c
#include <limits.h>
#include <stdlib.h>

/*@
    requires val > INT_MIN;
    ensures \result >= 0;
    ensures (val >= 0 ==> \result == val) && 
            (val < 0 ==> \result == -val);
*/
int my_abs(int val) {
    return abs(x);
}

/*@
    requires a > INT_MIN;
*/
void foo(int a) {
    int b = my_abs(-42);
    int c = my_abs(42);
    int d = my_abs(a);
}

File: transpiled_rust/frama_c_problems/absolute_value.rs
fn my_abs(val: i32) -> i32 {
    val.abs()
}

fn foo(a: i32) {
    let b: i32 = my_abs(-42);
    let c: i32 = my_abs(42);
    let d: i32 = my_abs(a);
}